src/sources/statsd/mod.rs

```diff
@@ -174,7 +177,10 @@ async fn statsd_udp(
         match frame {
             Ok((bytes, _sock)) => {
                 let packet = String::from_utf8_lossy(bytes.as_ref());
-                let metrics = packet.lines().filter_map(parse_event).map(Ok);
+                let metrics = packet
+                    .lines()
+                    .filter_map(|line| parse_event(Bytes::copy_from_slice(line.as_bytes())))
```

Comment:
It's a little funky that we're going `Bytes -> String -> lines() -> &str -> Bytes`.

If we replace the `BytesCodec` we're passing to `UdpFramed` with a `NewlineDelimitedCodec`, we should be able to simplify this a bit and avoid that extra copy/allocation.

Reply:
Oh man, good catch, that is hilariously convoluted 😄

Reply:
Changed in 91c45cc6eededea65c9aa26f005fded8cf7c0dbe